cameras/l515.cc

```diff
@@ -28,14 +28,21 @@ rs2_intrinsics L515::get_camera_intrinsics() const {
   return color_stream.get_intrinsics();
 }
 
-void L515::get_rgbd_frame(cv::Mat *color_img, cv::Mat *depth_img) const {
+int64_t L515::get_rgbd_frame(cv::Mat *color_img, cv::Mat *depth_img) const {
   auto frameset = pipe_.wait_for_frames();
   frameset = align_to_color_.process(frameset);
 
+  rs2::frame color_frame = frameset.get_color_frame();
+  rs2::frame depth_frame = frameset.get_depth_frame();
+
   *color_img = cv::Mat(cv::Size(WIDTH, HEIGHT), CV_8UC3, 
-      (void*)frameset.get_color_frame().get_data(), cv::Mat::AUTO_STEP);
+      (void*)color_frame.get_data(), cv::Mat::AUTO_STEP);
   *depth_img = cv::Mat(cv::Size(WIDTH, HEIGHT), CV_16UC1,
-      (void*)frameset.get_depth_frame().get_data(), cv::Mat::AUTO_STEP);
+      (void*)depth_frame.get_data(), cv::Mat::AUTO_STEP);
```

Comment:
I think we might need to explicitly copy things by calling `copyTo`. If `cv::Mat` is constructed with external pointer, the memory is not managed by `cv` and the pointer can go invalid without notice.

Reply:
Good call. But I think we should include this fix in the jetson_fix branch:

https://github.com/alvinsunyixiao/disinfect-slam/blob/alvin/jetson_fix/cameras/l515.cc